### ISSUE TITLE
OpenAI backend: logprobs is a boolean variable in the OpenAI API, use top_logprobs

### ISSUE THREAD

**Issue by sohaibimran7** [AISC10-team09/bayesian_reasoning#3]
From the OpenAI API:
<img width="658" alt="Image" src="https://github.com/user-attachments/assets/97f7f9cf-3491-466a-b2d5-b3421fa5b3d9" />

Ideally, the OpenAI backend should return an informative error if the required logits are not present in the 20 top_logprobs. The error should at least include the earlier tokens that were present and their cumulative logprobs, and the 20 top_logprobs (logits and logprobs) at the current step. 

Also, we may want to be consistent with the OpenAI API and use the term top_logprobs rather than logprobs to mean top_logprobs throughout.
